datalad/support/gitrepo.py

```diff
@@ -1266,6 +1266,60 @@ def _get_prefixed_commit_msg(msg):
         DATALAD_PREFIX = "[DATALAD]"
         return DATALAD_PREFIX if not msg else "%s %s" % (DATALAD_PREFIX, msg)
 
+    def for_each_ref_(self, fields=('objectname', 'objecttype', 'refname'),
+                      pattern=None, points_at=None, sort=None, count=None):
+        """Wrapper for `git for-each-ref`
+
+        Parameters
+        ----------
+        fields : list or str
```

Comment:
For most these parameters, I'm okay with not having a description because it is just a direct map to the `git for-each-ref` option.  But I think the description for `fields` should mention that it is used to construct the `--format` value.

Also, I suspect you're using a tuple as the default to avoid the bad practice/gotcha of using a mutable list as the default value (quickly scanning, I don't think a list would actually be problematic here, though it should still be avoided).  I'm fine with that (and would be fine with the more standard "None default, set list" as well), but I think if the default value is a tuple, the type shouldn't be documented as "list or str".  Perhaps "str or iterable of str"?

Reply:
I will add a description to all of them.

I used a tuple for those reasons, I used it over None to make it obvious from the signature what dict keys one can expect. I will fix the description.